[RNMobile] Add "Set as Featured" Button to Image Block

SiobhyB opened the pull request:
Paritial fix for: wordpress-mobile/gutenberg-mobile#1011

`gutenberg`: https://github.com/WordPress/gutenberg/pull/31415
`gutenberg-mobile`: https://github.com/wordpress-mobile/gutenberg-mobile/pull/3450

The main PR for the branch this individual PR will be merged into, `add/featured-functionality-to-image-block-ios`, can be found here: https://github.com/wordpress-mobile/WordPress-iOS/pull/16427

## Description

This PR will add a **Set as Featured** button to the image block's settings on iOS devices, with the purpose being to make it simpler for users to set a featured image within the post's editor. Users will also be able to **Remove as Featured** directly from the block's setting.

## Testing

### Main Test Flows

[The Gutenberg PR](https://github.com/WordPress/gutenberg/pull/31415) can be referred to as the central, most up-to-date repository for the main test flows. In addition, we should verify that the Tracks events edited/introduced with this PR work as expected.

### Tracks Verification

1. After following the steps to set and remove an image as featured directly from the image block, navigate to the internal Tracks tool and choose the **Live View** option from the menu.
2. Enter `wpios_editor_post_featured_image_changed` into the tool's **Events** field and then the username associated with the account you tested with in the **Username** field. Click **Search**.
3. Verify that the event was logged correctly. Note, there is usually a delay of ~5 minutes before an event is logged in this tool.
4. Click **Props** and scroll down to the **eventprops** section. The following props should be logged, depending on your actions:
-- **action** should either log `removed` or `added`, depending on whether you have removed or set a featured image.
-- **via** should log `gutenberg` when you set/remove a featured image via the image block or `settings` when this is done via Post Settings. 

## Regression Notes

1. Potential unintended areas of impact

As this PR adds another flow for users to set/remove featured images, there is some potential for it to unintentionally have a negative impact on the existing flow (found via Post Settings).

Another unintended area of impact is the `wpios_editor_post_featured_image_changed` Track event. This event currently only fires when a featured image is changed via Post Settings. With this PR, it should also be fired when a featured image is changed via the image block, with a `gutenberg` property to differentiate it.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

The steps taken to test whether the intended flow works as expected can be found in [the Gutenberg PR](https://github.com/WordPress/gutenberg/pull/31415) with additional steps for verifying the Tracks event fires as expected listed above.

3. What automated tests I added (or what prevented me from doing so)

N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

SiobhyB committed:
Pass ID of post's featured image over the bridge when editor first loads

SiobhyB committed:
Declare new "FeaturedImageDelegate" protocol

This protocol includes a method named didSetFeaturedImage, which will send a message to the JS side when a featured image is set from Post Settings.

SiobhyB committed:
Call didSetFeaturedImage when a featured image is set

SiobhyB committed:
Call didSetFeaturedImage in GutenbergViewController.swift

SiobhyB committed:
Establish connection between PostSettings and GutenbergViewController

SiobhyB committed:
Place didSetFeaturedImage() amongst other functions that send/receive data over the bridge

SiobhyB committed:
Set didSetFeaturedImage() to nil when featured image is removed

SiobhyB committed:
Remove trailing whitespace

SiobhyB committed:
Rename didSetFeaturedImage() to sendToJSFeaturedImageId()

The intention behind this change is to make the function's purpose clearer. It also moves the naming inline to this feature's Android counterpart. (See: https://github.com/wordpress-mobile/WordPress-Android/pull/14451.)

SiobhyB committed:
Renamed "sendToJSFeaturedImageId" to "gutenbergDidRequestFeaturedImageId"

This change is a bid align closer to the naming conventions of other functions in iOS.

SiobhyB committed:
Merge branch 'gutenberg/add/featured-functionality-to-image-block' into gutenberg/add/featured-badge

SiobhyB committed:
Add gutenbergDidRequestToSetFeaturedImage() function

This function will handle the logic for setting/removing a featured when the button is tapped.

SiobhyB committed:
Build out "setFeaturedImage" function

This function accepts a media ID for an image and uses the "existingMediaWith" to determine its media object, which can then be used to assign the image as the post's featured image. Lastly, the ID of the new featured image is then sent to Gutenberg via featuredImageIdNativeUpdated().

SiobhyB committed:
Build out notices to send when a new featured image is set

This commit includes different notices that will be sent to Gutenberg, depending on whether an image is being removed (which will be the case when the media ID is zero) or set.

SiobhyB committed:
Add showAlertForReplacingFeaturedImage() function

This function will be called when a post already has a featured image and will prompt the user on whether they wish to replace the current featured image or not.

SiobhyB committed:
Build out logic in gutenbergDidRequestToSetFeaturedImage() method

This commit introduces logic into the gutenbergDidRequestToSetFeaturedImage() function. Specifically, an if/else statement is used to either set, remove, or show an alert to confirm a replacement, depending on the state of featured images in the post.

SiobhyB committed:
Build out alertController for showAlertForReplacingFeaturedImage()

SiobhyB committed:
Remove trailing whitespace

SiobhyB committed:
Tidy up if/else logic in gutenbergDidRequestToSetFeaturedImage()

SiobhyB committed:
Merge branch 'gutenberg/add/featured-functionality-to-image-block' into gutenberg/add/set-as-featured-button

SiobhyB committed:
Merge branch 'gutenberg/add/featured-functionality-to-image-block' into gutenberg/add/set-as-featured-button

SiobhyB committed:
Merge branch 'gutenberg/add/featured-functionality-to-image-block' into gutenberg/add/set-as-featured-button

SiobhyB committed:
New 'GutenbergFeaturedImageHelper' class

With this commit, the setFeaturedImage() function is moved away from 'GutenbergMediaInserterHelper' and to a new  'GutenbergFeaturedImageHelper' class. The code serves a slightly separate purpose to the other code in 'GutenbergMediaInserterHelper', so the aim of moving it to a new class is to improve overall readability.

References/calls to the old class/function are also updated in GutenbergViewController.swift as part of this commit, to ensure the 'set featured' functionality continues to work.

SiobhyB committed:
Refactor if/else statement for readability

With this commit, the if/else statement is refactored to a series of single if statements, with accompanying comments. The goal is to make the code more readable and for it to be clearer what the purpose of each combination is. It also adds an explicit if statement for any cases where the media ID may be equal to the featured image's ID.

This brings the code inline to that used for the Android side of this feature. See: https://github.com/wordpress-mobile/WordPress-Android/pull/14503/commits/fe93e0b88405bd4240c6263769ed56fae72f14d0

SiobhyB committed:
Introduce 'mediaIdNoFeaturedImageSet' constant

With this commit, a 'mediaIdNoFeaturedImageSet' constant is introduced and used in checks where a featured image is being removed.

The purpose behind introducing a constant is to improve readability and make it easier to change in the future, if needed.

This brings the code inline with the Android side of the functionality. See: https://github.com/wordpress-mobile/WordPress-Android/pull/14503/commits/9630c2955da873dbbd033849a6d1494876bbb582

SiobhyB committed:
Address styling errors reported by Hound

Hound reported the following two styligng errors in several places in the code, which are resolved with this commit:

- Trailing Whitespace Violation: Lines should not have trailing whitespace. (trailing_whitespace)
-  Control Statement Violation: if, for, guard, switch, while, and catch statements shouldn't unnecessarily wrap their conditionals or arguments in parentheses. (control_statement)

SiobhyB committed:
Fire tracks event when image is removed/added

The existing tracks event for a featured image change (editorPostFeaturedImageChanged) is utilised, with its properties updated to reflect that the change was made via the Gutenberg editor (rather than via Post Settings).

SiobhyB committed:
Unwrap 'featuredImageId' to prevent crash when nil

When the featured image's ID was nil (that is, when there was no featured image) and an attempt was made to set a featured image, the app crashed with the following error: "force unwrapped a nil value"

The crash is fixed with this commit, by checking whether the 'featuredImageId' is nil in the first check within the 'gutenbergDidRequestToSetFeaturedImage' function.

With this commit, a new 'featuredImageId' has also been created, as its value is now used in three places in the code block.

SiobhyB committed:
Reposition 'mediaIdNoFeaturedImageSet' constant

With this commit, the constant has been moved from the top of the file to inside the 'gutenbergDidRequestToSetFeaturedImage' function where it's used.

The constant was repositioned in a bid to improve the overall readability of the code and to follow the logic I'm seeing elsewhere in the class (I'm not seeing a pattern of constants at the top of the file).

SiobhyB committed:
Edit descriptions of strings for clarity

The existing descriptions accompanying strings have been edited with this commit, with a view to make them as clear as possbile for translators.

Two descriptions have also been added to strings that didn't have them ("replace" and "cancel").

Finally, the "Replace current?" string have been updated to "Replace" in a bid to make it more translateable.

SiobhyB committed:
Tweaked wording of comment for clarity

The comment noting the reason for dismissing the current controller alluded to "controller behind settings modal", which is a bit wordy and may be confusing. I simplified the language in order to hopefully be clearer while still conveying the reason for the dismissal.

SiobhyB committed:
Reference latest Gutenberg Mobile commit

SiobhyB committed:
Hash and pod changes

SiobhyB committed:
Add GutenbergFeaturedImageHelper to project.pbxproj

SiobhyB commented:
👋 @guarani, I was wondering whether you may have any availability to review this PR this coming week? 

The code is ready for review, but reliant on [the Android PR](https://github.com/WordPress/gutenberg/pull/31705) being merged first, which is why the chain of PRs are still set as drafts. For that reason, [the Gutenberg PR](https://github.com/WordPress/gutenberg/pull/31415) still includes some files that will eventually be merged with that Android one. I've listed the ones that are specific to iOS (and will therefore need to be reviewed as part of this iOS chain) in the PR's description.

As the Android PR is still pending a final review (and I'm not sure when that will happen), this isn't urgent and there's no worries if you're swamped. I just thought I'd check in to see if you might have availability for reviewing some time this week, as you helped me a lot with the previous iOS one. :)

guarani commented:
👋 @SiobhyB, sure, feel free to add me as a reviewer!

SiobhyB commented:
Thank you so much, I've assigned you now. 🙇‍♀️

guarani commented on WordPress/Classes/ViewRelated/Gutenberg/GutenbergFeaturedImageHelper.swift:
```diff
@@ -0,0 +1,37 @@
+        post.featuredImage = media
+
+        if mediaID == mediaIdNoFeaturedImageSet {
+            gutenberg.showNotice(NSLocalizedString("Removed as Featured Image", comment: "Notice confirming that an image has been remove as the post's featured image."))
```
```suggestion
            gutenberg.showNotice(NSLocalizedString("Removed as Featured Image", comment: "Notice confirming that an image has been removed as the post's featured image."))
```

guarani commented on WordPress/Classes/ViewRelated/Gutenberg/GutenbergViewController.swift:
```diff
@@ -648,6 +653,49 @@ extension GutenbergViewController: GutenbergBridgeDelegate {
+        }
+
+        alertController.addAction(replaceAction)
+        alertController.addCancelActionWithTitle(NSLocalizedString("Replace", comment: "Button to cancel the replacement of a featured image."))
```
```suggestion
        alertController.addCancelActionWithTitle(NSLocalizedString("Cancel", comment: "Button to cancel the replacement of a featured image."))
```

SiobhyB replied:
Thanks for spotting this! I've gone ahead to apply the suggestion.

guarani commented on WordPress/Classes/ViewRelated/Gutenberg/GutenbergViewController.swift:
```diff
@@ -648,6 +653,49 @@ extension GutenbergViewController: GutenbergBridgeDelegate {
 
+    func gutenbergDidRequestToSetFeaturedImage(for mediaID: Int32) {
+        // dismiss current controller in order to allow the new alert controller to be presented
+        presentedViewController?.dismiss(animated: false, completion: nil)
```
Is this to remove any view controller that might be currently presented, a sort of catch-all? If so, the only thing that I'm concerned about is that its dismissing without an animation which could be jarring to users. 

Would putting all the action sheet code inside the completion handler above work? I don't recall if it's called even if there's nothing to dismiss though, but it might be worth checking if you haven't already.

SiobhyB replied:
@guarani, the bottom sheet modal (containing the image block's settings) is currently dismissed in the JS [here](https://github.com/WordPress/gutenberg/blob/53013060c4bbda4b79bff116f80a226322de47fd/packages/block-library/src/image/edit.native.js#L690) as soon as the user taps the `Set/Remove as Featured` button. But that doesn't seem to be being automatically "communicated" back to the iOS side, which is what led me adding this line of code. If I remove that line, the replace functionality fails with the following error:

`Attempt to present <UIAlertController: 0x7ff44aa16400> on <WordPress.AztecNavigationController: 0x7ff44a068200> (from <WordPress.GutenbergViewController: 0x7ff44a808e00>) which is already presenting <RCTModalHostViewController: 0x7ff437eeb340>.`

I found the open issue you created at https://github.com/WordPress/gutenberg/issues/29672 that seems to explain the reason for this behaviour but wasn't certain of the correct way to go about addressing this in my case, where the a new dialog is generated from the iOS side. Does the approach I've taken make more sense to you now with that extra context or is there something else that should be done here?

> Would putting all the action sheet code inside the completion handler above work? I don't recall if it's called even if there's nothing to dismiss though, but it might be worth checking if you haven't already.

Just to make sure I understand: Are you imagining something like this?

`presentedViewController?.dismiss(animated: false, completion: {self.showAlertForReplacingFeaturedImage(mediaID: mediaID)})
`

If so, I tested adding that at the bottom of the `gutenbergDidRequestToSetFeaturedImage` function and it works as expected. :) Would this be a better approach to take, as there will always be the bottom sheet that needs to be dismissed? 

Is there a chance that the way bottom sheets are dismissed may change in the future and negate the need for this extra dismissal in iOS (therefore break the functionality if we're relying on the completion handler)?

guarani replied:
> But that doesn't seem to be being automatically "communicated" back to the iOS side, which is what led me adding this line of code. If I remove that line, the replace functionality fails with the following error:

True, this sounds familiar and thanks for highlighting that existing issue. I was thinking something along the lines of this:

```
    func gutenbergDidRequestToSetFeaturedImage(for mediaID: Int32) {

        let presentAlert = { [weak self] in
            guard let `self` = self else { return }

            // ... the various guard-statements go here ...

            // ask the user to confirm changing the featured image since there's already one set
            self.showAlertForReplacingFeaturedImage(mediaID: mediaID)
        }

        if presentedViewController != nil {
            dismiss(animated: true, completion: presentAlert)
        } else {
            presentAlert()
        }
    }
```

I think this may work now and also if the bottom sheet behavior is fixed in the future, wdyt?

SiobhyB replied:
Thank you so much for sharing that with me, I can see how it's a more eloquent solution and have gone ahead to implement it in 89494c318e69c07418b2f69781e2686ce2fc3e72.

Just a note: I kept `animated: false` as setting this value to `true` caused a crash and the animation is still handled on the JS side when the BottomSheet's dismissed.

Let me know if you spot any further issues here!

guarani replied:
Looks good! Testing now and things seem to work well 👍

guarani commented on WordPress/Classes/ViewRelated/Gutenberg/GutenbergViewController.swift:
```diff
@@ -648,6 +653,49 @@ extension GutenbergViewController: GutenbergBridgeDelegate {
+        presentedViewController?.dismiss(animated: false, completion: nil)
+
+        let featuredImageId = post.featuredImage?.mediaID
+        let mediaIdNoFeaturedImageSet = 0
```
This seems to be declared in two places ([this](https://github.com/wordpress-mobile/WordPress-iOS/pull/16426/files#diff-1f88390d90a5eb0598cd4101f287ac4468bfe3229fa20eb41e78d386efc6fa28R8) being the other).
Would it make sense to put this in a static constant inside `GutenbergFeaturedImageHelper`, like [this](https://github.com/wordpress-mobile/WordPress-iOS/blob/6bd1fbfeae283cc43e04acbf40386b87051fe4aa/WordPress/Classes/ViewRelated/User%20Profile%20Sheet/UserProfileSiteCell.swift#L11)?

SiobhyB replied:
Ah, that makes a lot of sense, I can see how that would be better practice! I've gone ahead to make that change in https://github.com/wordpress-mobile/WordPress-iOS/pull/16426/commits/9d1aaf599e76370edcb0fedcc3d1077e9e8ee590. 🙇‍♀️

guarani commented on WordPress/Classes/ViewRelated/Gutenberg/GutenbergViewController.swift:
```diff
@@ -648,6 +653,49 @@ extension GutenbergViewController: GutenbergBridgeDelegate {
+        let featuredImageId = post.featuredImage?.mediaID
+        let mediaIdNoFeaturedImageSet = 0
+
+        if featuredImageId != nil && mediaID == featuredImageId as! Int32 {
+            // nothing special to do, trying to set the image that's already set as featured
+            return
+        }
```
If the goal here is to exit the function if `featuredImageId` is equal to `mediaID`, then you could consider using Swift's guard-statement. It's often used to abort a function if certain conditions are not met:
```
        guard featuredImageId as? Int32 != mediaID else {
            return
        }
```

SiobhyB replied:
Thank you for this! I've gone ahead to refactor the if statements to use guard in https://github.com/wordpress-mobile/WordPress-iOS/pull/16426/commits/25da33ea7dd83766704ec42ef6d067e554a86be0 and will know to consider using guard if I do any further iOS work in the future. :D

guarani commented on WordPress/Classes/ViewRelated/Gutenberg/GutenbergViewController.swift:
```diff
@@ -648,6 +653,49 @@ extension GutenbergViewController: GutenbergBridgeDelegate {
+            return
+        }
+
+        if mediaID == mediaIdNoFeaturedImageSet {
+            // user tries to clear the featured image setting
+            featuredImageHelper.setFeaturedImage(mediaID: mediaID)
+            return
+        }
```
Again you could consider using the guard-statement. Note that the condition is negated when using it (i.e. only continue if the media ID _is not_ zero, otherwise do some clean up and exit):
```
        guard mediaID != mediaIdNoFeaturedImageSet else {
            // user tries to clear the featured image setting
            featuredImageHelper.setFeaturedImage(mediaID: mediaID)
            return
        }
```

guarani commented on WordPress/Classes/ViewRelated/Gutenberg/GutenbergViewController.swift:
```diff
@@ -648,6 +653,49 @@ extension GutenbergViewController: GutenbergBridgeDelegate {
+            return
+        }
+
+        if featuredImageId == nil {
+            // current featured image is not set so, go ahead and set it to the provided one
+            featuredImageHelper.setFeaturedImage(mediaID: mediaID)
+            return
+        }
```
Here a guard-statement could be used instead, since we're again aborting if conditions are not met.

guarani commented on WordPress/Classes/ViewRelated/Gutenberg/GutenbergFeaturedImageHelper.swift:
```diff
@@ -0,0 +1,37 @@
+    fileprivate let post: AbstractPost
+    fileprivate let gutenberg: Gutenberg
+
+    let mediaIdNoFeaturedImageSet = 0
```
As noted in other comments below, this could be a static let instead, allowing you to define it once here and easily use it in other files.

guarani reviewed (commented):
Looking great! Your Swift is coming along nicely :D I left a few comments, but for the most part it's working well. Feel free to re-request a review when this is ready again.

SiobhyB committed:
Fix typo in string's description

With this commit, 'remove' is updated to 'removed' in a string's description, resolving a typo.

Co-authored-by: Paul Von Schrottky <paul.von.schrottky@automattic.com>

SiobhyB committed:
Fix typo in string

'Replace' was used where 'Cancel' should have been, which is resolved with this commit.

Co-authored-by: Paul Von Schrottky <paul.von.schrottky@automattic.com>

SiobhyB committed:
Convert variable into static constant

With this commit, the 'mediaIdNoFeaturedImageSet' variable, which was being declared in two separate files, is converted into a static constant. This will make the value easier to change in the future and make it more re-usable.

SiobhyB committed:
Remove trailing white space

SiobhyB committed:
Convert if statements to guard statements

SiobhyB committed:
Remove trailing whitespace

SiobhyB commented:
Thank you so much for this, Paul. 🙇‍♀️ I've gone ahead to implement the changes you've suggested and had a question on one of the suggestions [here](https://github.com/wordpress-mobile/WordPress-iOS/pull/16426#discussion_r644000484).

SiobhyB committed:
Reword action sheet's title for clarity

The title of the replace dialog is updated from "Replace?" to "Featured Image Already Set" with this commit. This brings it more inline with how titles are presented in other areas of the app (questions aren't generally used for titles). The overall goal is to improve clarity.

SiobhyB committed:
Undo title case from notices for consistency

Following feedback, it makes sense to undo title case that's applied to notices, as they're not strictly headings/titles. This will help with consistency in how notices are capitalised throughout the rest of the apps.

SiobhyB committed:
Update Gutenberg Mobile reference

SiobhyB committed:
Podfile changes

guarani commented:
Just a heads-up that the unit tests seem to be failing and I'm seeing an error when the block editor opens. Once this is done I can test the Tracks events which I'd forgotten to do in the first review.

<img src="https://user-images.githubusercontent.com/1898325/120561087-edcabc80-c3d1-11eb-949d-c422e9163e60.jpg" alt="" width="350">

SiobhyB committed:
Merge branch 'gutenberg/add/featured-functionality-to-image-block' into gutenberg/add/set-as-featured-button

SiobhyB committed:
Podfile changes

SiobhyB committed:
Podfile changes

SiobhyB committed:
Merge branch 'gutenberg/add/featured-functionality-to-image-block' into gutenberg/add/set-as-featured-button

SiobhyB committed:
Refactor logic around dismissing controller

It's necessary to dismiss then present controller at the moment due to an issue with BottomSheet. See hee for further discussion: https://github.com/wordpress-mobile/WordPress-iOS/pull/16426/files#r643544481

With this commit, the logic around dismissing the controller is refactored to make use of the completion handler.

An if/else statment is used to account for any cases where a controller isn't present (i.e. if the issue with BottomSheet is fixed in the future).

SiobhyB committed:
Remove trailing whitespace

SiobhyB commented:
@guarani, sorry for missing that error. 🤦‍♀️   

I've double-checked and everything should work without error with the latest changes on this branch now.

guarani commented:
> ### Tracks Verification

Tested 287307b08c54036db7bf2c0babe4ee2beeac74cf:

- [x] Tracks events logged correctly when adding and removing a feature image via the Image block
- [x] Tracks events logged correctly when adding and removing a feature image via Post Settings

SiobhyB committed:
Podfile and Podfile.lock changes

SiobhyB committed:
Make 'replace' dialog work with iPad's popovers

iPads display action sheets inside a popover, which wasn't accounted for in the code, leading to a crash when the 'replace featured' dialog was called.

The crash is fixed with this commit, which introduces support for iPad's popovers.

SiobhyB committed:
Merge branch 'gutenberg/add/featured-functionality-to-image-block' into gutenberg/add/set-as-featured-button

SiobhyB committed:
Refactor code for iPad-specific popoverController

This commit refactors the code for iPad-specific popoverController. It's now more inline with similar code elsewhere in the GutenbergViewController.swift file, for example here:

https://github.com/wordpress-mobile/WordPress-iOS/blob/develop/WordPress/Classes/ViewRelated/Gutenberg/GutenbergViewController.swift#L746

SiobhyB committed:
Remove trailing whitespace

SiobhyB committed:
Podfile changes

SiobhyB committed:
Podfile changes

SiobhyB commented:
@guarani, this is ready for the next review when you have a chance. :) I've updated all the PRs in this chain so that they're up-to-date following HACK week and my recent AFK. The most recent changes also include [a fix for the iPad-specific crash you surfaced](https://github.com/wordpress-mobile/WordPress-iOS/pull/16426/commits/998e411936f10df080e335a5b3d7e9fd19f58b52).

SiobhyB committed:
retrigger checks

SiobhyB commented:
Highlighting that I found an issue where [the button breaks the editor's undo/redo functionality](https://github.com/WordPress/gutenberg/issues/32937), which I'll work on in a separate issue. I'd like to propose merging this issue into the main feature branch (if no other issues are found in this PR chain) irrespective of that particular known issue, as it'll be worked on separately to this.

guarani reviewed (commented):
I tested this on an iPad and the replace action sheet is being shown right at the top of the screen which looks incorrect.


https://user-images.githubusercontent.com/1898325/123307750-dd998f00-d4f0-11eb-83c3-f44f328db66b.MP4

SiobhyB committed:
Center and re-style popoverController

With this commit, the iPad-specific popoverController is centred in the middle of the screen. The "permittedArrowDirections" value is also changed to remove any arrow and ensure the dialog's message is displayed in full (rather than only the title).

SiobhyB commented:
Thanks @guarani, I've gone ahead to make some changes in https://github.com/wordpress-mobile/WordPress-iOS/pull/16426/commits/8f473649cfb790a736257fb742e8bfea7c31adc5 so that the dialog is always centred in the screen. I also removed the arrow so that the message is displayed (rather than only the title, which seems to happen when an arrow is specified), which I think looks a bit better in this case:

![Simulator Screen Shot - iPad Pro (12 9-inch) (4th generation) - 2021-06-24 at 21 48 22](https://user-images.githubusercontent.com/2998162/123330433-eb253800-d535-11eb-966f-a4aa2a93c9fb.png)

Let me know what you think!

guarani reviewed (approved):
Code looks great and works as expected! 💯 
Approved via https://github.com/WordPress/gutenberg/pull/31415#pullrequestreview-692255299
Thanks @SiobhyB!
